Sync/Extractor/DoctrineExtractor.php

```diff
@@ -66,18 +68,24 @@ public function extract(BaseDiffItem $item)
 
                 $idFieldName = str_replace(['OLD.', 'NEW.'], '', $idField);
                 $itemRow = $item->getItem();
-                $itemId = $itemRow[$idFieldName];
-
-                $storage = $this->getStorageFacility();
-                $storage->save(
-                    $action,
-                    $insertList[JobTableFields::TYPE]['value'],
-                    $itemId,
-                    $item->getTimestamp(),
-                    $this->getShopIds()
-                );
 
                 $statements = $relation->getStatements();
+
+                if (isset($insertList[JobTableFields::TYPE]['value'])) {
+                    $itemId = $itemRow[$idFieldName];
+
+                    $storage = $this->getStorageFacility();
+                    $storage->save(
+                        $action,
+                        $insertList[JobTableFields::TYPE]['value'],
+                        $itemId,
+                        $item->getTimestamp(),
+                        $this->getShopIds()
+                    );
+                } elseif (empty($statements)) {
+                    throw new \LogicException('Relation does not have any effect');
```

Comment:
This happens only when relation does not mark item as changed and has no related items so the cause of this is not always empty relation.